rhtconsulting/rhc-ose
# Introducing support for OSEv3.1 to the deployment tools

#### What does this PR do?

Changes made to the rhc-ose tools to allow for installation of OSEv3.1. It also allows choosing which version to install - i.e.: 3.0.x or 3.1.x.

Please note the following:
- The base images have been renamed to "ose3_X-base" where X is the minor number. This means that testing this PR will not interfere with already running 3.0.x systems. 
- There are multiple files that have been named 30 v.s 31, this is to allow for an easier customization of the new version without impacting the existing 3.0 functionality part of the tool
#### How should this be manually tested?

Several steps to test - from a high level:
1. Generate new base images for 3.0 v.s. 3.1
2. Use images from step 1 to install OSEv3.0 and a second env for OSEv3.1

**NOTE:** There are most likely more testing needed than what's outlined here, but these are the basic steps that should be executed.

To generate new images, run:

```
cd provisioning/openstack
./create_base_image -k=<OS1_key> -v=3.0 # to generate a base image for OSEv3.0
./create_base_image -k=<OS1_key> -v=3.1 # to generate a base image for OSEv3.1
```

Promote the images to become the new base images:

```
cd provisioning
./osc-manage-image --action=promote --image-name=<3.0_name_from_prev_step> --ose-version=3.0 
./osc-manage-image --action=promote --image-name=<3.1_name_from_prev_step> --ose-version=3.1 
```

Next, use the newly created images to deploy new environments

```
cd provisioning
./osc-provision --num-nodes=2 --key=<OS1_key> # this will deploy the default (3.0)
./osc-provision --num-nodes=2 --key=<OS1_key> --ose-version=3.1 # this will deploy OSEv3.1
```

In the newly created environments, among other things, validate:
- the environment is running :-)
- the nodes have successfully connected to the master (oc get nodes)
- the registry and router have been successfully deployed
- the routes have been correctly configured
- the basic users can login (i.e.: joe and alice)
- can use the environment to perform a S2I deployment
#### Is there a relevant Issue open for this?

N/A - however, this takes care of issue #71 
#### Who would you like to review this?

/cc @etsauer
